subgraphs/rocket-pool/src/updaters/financialMetrics.ts

```diff
@@ -0,0 +1,461 @@
+import {
+  Address,
+  BigInt,
+  BigDecimal,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import { bigIntToBigDecimal } from "../utils/numbers";
+import { getOrCreateProtocol } from "../entities/protocol";
+import { getOrCreatePool } from "../entities/pool";
+import {
+  FinancialsDailySnapshot,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+} from "../../generated/schema";
+import { getOrCreateToken } from "../entities/token";
+import {
+  BIGDECIMAL_ZERO,
+  ETH_ADDRESS,
+  RETH_ADDRESS,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+  RPL_ADDRESS,
+} from "../utils/constants";
+
+const PROTOCOL_ID = RETH_ADDRESS;
+
+export function updateProtocolAndPoolRewardsTvl(
+  blockNumber: BigInt,
+  blockTimestamp: BigInt,
+  rewardAmount: BigInt
+): void {
+  const pool = getOrCreatePool(blockNumber, blockTimestamp);
+  const protocol = getOrCreateProtocol();
+
+  const inputTokenBalances: BigInt[] = [];
+  inputTokenBalances.push(rewardAmount);
+  inputTokenBalances.push(pool.inputTokenBalances[1]);
+
+  pool.inputTokenBalances = inputTokenBalances;
+
+  // inputToken is ETH, price with ETH
+
+  const ethTVLUSD = bigIntToBigDecimal(inputTokenBalances[1]).times(
+    getOrCreateToken(Address.fromString(ETH_ADDRESS), blockNumber).lastPriceUSD!
+  );
+
+  const rplTVLUSD = bigIntToBigDecimal(inputTokenBalances[0]).times(
+    getOrCreateToken(Address.fromString(RPL_ADDRESS), blockNumber).lastPriceUSD!
+  );
+
+  const totalValueLockedUSD = ethTVLUSD.plus(rplTVLUSD);
```

Comment:
Also the staked RPL should be a separate Pool

Reply:
@bye43 can you confirm this is right? Since RPL is required to be staked alongside nodes it kinda seems like it should be TVL to me, but you are the LSD expert now 🌈

Reply:
TVL doesn't include staked RPL? I don't think this is what was originally outlined. and I think it definitely should be. Esp because that tracks slashed RPL etc.

Reply:
In the future, I think that we should have staked RPL and ETH as separate pools since it makes comparison across the industry standardized. But for now, counting RPL as part of TVL is fine.